### PR TITLE
Fixed TryReadMessage() throwing exceptions

### PR DESCRIPTION
**Pull request by ThomasMiz**
Fixes issue #3

### CODE REVIEW

**Comment by ThomasMiz** [comment]
@TechnologicalPizza Can we get a comment on these changes?

File: Lidgren.Network/Connection/NetConnection.cs [hunk 1]
```diff
@@ -131,7 +131,7 @@ internal void SetStatus(NetConnectionStatus status, string? reason = null)
                 info.EnsureCapacity(4 + reason.Length + (reason.Length > 126 ? 2 : 1));
                 info.SenderConnection = this;
                 info.SenderEndPoint = RemoteEndPoint;
-                info.Write(Status);
+                info.Write((byte)Status);
```
Comment (TechPizzaDev) [inline Lidgren.Network/Connection/NetConnection.cs]: It should be written as a variable length Int64 as it's an enum. That may indicate that other parts of the library still expect a byte.

**Comment by TechPizzaDev** [comment]
How strange, didn't get a notification of the issue before.
Regardless, the NetQueue change seems appropriate but I'm confused about the status part. Added a comment on it.

**Comment by ThomasMiz** [comment]
Alright, so according to the Lidgren.Network documentation (or at least [this one](https://code.google.com/archive/p/lidgren-network-gen3/wikis/IncomingMessageTypes.wiki)), StatusChanged messages start with a single byte which can be cast to a NetConnectionStatus to know to what status the NetConnection has changed. However, the code previous to this PR was writting an enum, which I'm not sure how it got interpreted but was definitely not a single byte.

**Comment by TechPizzaDev** [comment]
https://github.com/TechnologicalPizza/lidgren-network-gen3/issues/3#issuecomment-753642697
I should've written that here

**Pull request merged**